insonmnia/locator/server.go

```diff
@@ -1,183 +1,58 @@
+	log.G(ctx).Info("Handling Announce request",
 		zap.Stringer("eth", ethAddr), zap.Strings("ips", req.IpAddr))
 
-	l.putAnnounce(&node{
-		ethAddr: ethAddr,
-		ipAddr:  req.IpAddr,
+	err = s.s.Put(&ds.Node{
+		EthAddr: ethAddr,
+		IpAddr:  req.IpAddr,
 	})
 
-	return &pb.Empty{}, nil
+	return &pb.Empty{}, err
 }
 
-func (l *Locator) Resolve(ctx context.Context, req *pb.ResolveRequest) (*pb.ResolveReply, error) {
-	log.G(l.ctx).Info("handling Resolve request", zap.String("eth", req.EthAddr))
+func (s *Srv) Resolve(ctx context.Context, req *pb.ResolveRequest) (*pb.ResolveReply, error) {
+	log.G(ctx).Info("Handling Resolve request", zap.String("eth", req.EthAddr))
 
 	if !common.IsHexAddress(req.EthAddr) {
 		return nil, fmt.Errorf("invalid ethaddress %s", req.EthAddr)
 	}
 
-	n, err := l.getResolve(common.HexToAddress(req.EthAddr))
-	if err != nil {
-		return nil, err
-	}
-
-	return &pb.ResolveReply{IpAddr: n.ipAddr}, nil
-}
-
-func (l *Locator) Serve() error {
-	lis, err := net.Listen("tcp", l.conf.ListenAddr)
-	if err != nil {
-		return err
-	}
-
-	return l.grpc.Serve(lis)
-}
-
-func (l *Locator) extractEthAddr(ctx context.Context) (common.Address, error) {
-	pr, ok := peer.FromContext(ctx)
-	if !ok {
-		return common.Address{}, status.Error(codes.DataLoss, "failed to get peer from ctx")
-	}
-
-	switch info := pr.AuthInfo.(type) {
-	case util.EthAuthInfo:
-		return info.Wallet, nil
-	default:
-		return common.Address{}, status.Error(codes.Unauthenticated, "wrong AuthInfo type")
-	}
-}
-
-func (l *Locator) putAnnounce(n *node) {
-	l.mx.Lock()
-	defer l.mx.Unlock()
-
-	n.ts = time.Now()
-	l.db[n.ethAddr] = n
-}
-
-func (l *Locator) getResolve(ethAddr common.Address) (*node, error) {
-	l.mx.Lock()
-	defer l.mx.Unlock()
-
-	n, ok := l.db[ethAddr]
-	if !ok {
-		return nil, errNodeNotFound
-	}
-
-	return n, nil
-}
-
-func (l *Locator) cleanExpiredNodes() {
-	t := time.NewTicker(l.conf.CleanupPeriod)
-	defer t.Stop()
-
-	for {
-		select {
-		case <-t.C:
-			l.traverseAndClean()
-		}
-	}
-}
-
-func (l *Locator) traverseAndClean() {
-	deadline := time.Now().Add(-1 * l.conf.NodeTTL)
-
-	l.mx.Lock()
-	defer l.mx.Unlock()
-
-	var (
-		total = len(l.db)
-		del   uint64
-		keep  uint64
-	)
-	for addr, node := range l.db {
-		if node.ts.Before(deadline) {
-			delete(l.db, addr)
-			del++
-		} else {
-			keep++
-		}
-	}
-
-	log.G(l.ctx).Debug("expired nodes cleaned",
-		zap.Int("total", total), zap.Uint64("keep", keep), zap.Uint64("del", del))
-}
-
-func NewLocator(ctx context.Context, conf *LocatorConfig, key *ecdsa.PrivateKey) (l *Locator, err error) {
-	if key == nil {
-		return nil, errors.Wrap(err, "private key should be provided")
-	}
-
-	l = &Locator{
-		db:     make(map[common.Address]*node),
-		conf:   conf,
-		ctx:    ctx,
-		ethKey: key,
-	}
-
-	var TLSConfig *tls.Config
-	l.certRotator, TLSConfig, err = util.NewHitlessCertRotator(ctx, l.ethKey)
+	n, err := s.s.ByEthAddr(common.HexToAddress(req.EthAddr))
```

Comment:
s.s. - "Schutzstaffel" ?

Reply:
I beg you pardon, but I don't speak German. Would you be so kind as to elucidate on the profound though stated above?